File: visitor/java/Book.java
package Visitor;

// Temel sınıftır.
// Yeni davranışları Accept metotu ile alır.
// UML diyagramındaki Element yapısına denk gelmektedir.
abstract class Book
{
    public String name;
    public String author;
    public String categoryName;

    public abstract void accept(IBookExportVisitor visitor);
}

File: visitor/python/book.py
from abc import ABC, abstractmethod


class Book(ABC):
    """
    Temel sınıftır.
    Yeni davranışları Accept metotu ile alır.
    UML diyagramındaki Element yapısına denk gelmektedir.
    """
    name: str
    author: str
    category_name: str

    @abstractmethod
    def accept(self, visitor):
        pass
